sn_networking/src/lib.rs

```diff
@@ -621,6 +622,43 @@ impl Network {
             .await)
     }
 
+    pub async fn get_store_cost_from_network(&self, dbc_id_key: NetworkAddress) -> Result<Token> {
+        let (sender, receiver) = oneshot::channel();
+        debug!("Attempting to get store cost");
+        // first we need to get CLOSE_GROUP of the dbc_id
+        self.send_swarm_cmd(SwarmCmd::GetClosestPeers {
+            key: dbc_id_key.clone(),
+            sender,
+        })?;
+
+        let close_nodes = receiver
+            .await
+            .map_err(|_e| Error::InternalMsgChannelDropped)?
+            .into_iter()
+            .collect_vec();
+
+        let request = Request::Query(Query::GetStoreCost(dbc_id_key));
```

Comment:
we already sending the Query to the specific peers.
Hence the `address` of the target is not that matter, instead the `size` will be more interested?

Reply:
I'm not sure what you mean here

Reply:
I mean for the `query` request, we already `get_clostest_peers` within the above step according to the `dbc_id_key` address, and then send the request directly to peers one by one.
Hence, the `address content` with the `Query::GetStoreCost` doesn't matter anymore. Unless the cost is related to the data size to be uploaded, then what shall be wrapped in the Query shall be `data size` instead of `data address`.

Reply:
Ah right. Yeh we can drop that entirely I think. the price is per record, so size doesn't matter to nodes, client can work it out.